contracts/token/ERC721/extensions/ERC721Wrapper.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "../utils/ERC721Holder.sol";
+
+/**
+ * @dev Extension of the ERC721 token contract to support token wrapping.
+ *
+ * Users can deposit and withdraw an "underlying token" and receive a "wrapped token" with a matching tokenId. This is useful
+ * in conjunction with other modules. For example, combining this wrapping mechanism with {ERC721Votes} will allow the
+ * wrapping of an existing "basic" ERC721 into a governance token.
+ *
+ * _Available since v4.9.0_
+ */
+abstract contract ERC721Wrapper is ERC721, ERC721Holder {
+    IERC721 public immutable underlying;
```

Comment:
All state variables are private. This seems fine if it's immutable, but in the upgradeable version it will become a storage variable so we would want it to be private so it can't be set by children contracts.

Reply:
Moved and [created an issue for ERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/4026)